aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -96,41 +97,105 @@ Completable hydrate() {
 
         // Observe the remote model states,
         // Get an observable stream of the set of model classes.
+        if (modelProvider.models() == null || modelProvider.models().isEmpty()) {
+            return hydrateSerializedModels(modelWithMetadataComparator);
+        } else {
+            return hydrateNonSerializedModels(modelWithMetadataComparator);
+        }
```

Comment:
Splitif we have access to models (typical android usage) or modelSchemas only (request from Flutter)

Reply:
No longer needed.